app/components/map-component/index.tsx

```diff
@@ -0,0 +1,201 @@
+import MapMarkerComponent from "../map-marker-component"
+import { PermissionStatus, RESULTS, request } from "react-native-permissions"
+import { LOCATION_PERMISSION, getUserRegion } from "@app/screens/map-screen/map-screen"
+import LocationButtonCopy from "./location-button-copy"
+import debounce from "lodash.debounce"
+import { updateMapLastCoords } from "@app/graphql/client-only-query"
+import { useApolloClient } from "@apollo/client"
+import { OpenSettingsElement, OpenSettingsModal } from "./open-settings-modal"
+import { View } from "react-native"
+import { isIOS } from "@rneui/base"
+
+type Props = {
+  data?: BusinessMapMarkersQuery
+  userLocation?: Region
+  permissionsStatus?: PermissionStatus
+  setPermissionsStatus: (_: PermissionStatus) => void
+  handleMapPress: () => void
+  handleMarkerPress: (_: MapMarker) => void
+  focusedMarker: MapMarker | null
+  focusedMarkerRef: React.MutableRefObject<MapMarkerType | null>
+  handleCalloutPress: (_: MapMarker) => void
+  alertOnLocationError: () => void
+}
+
+export default function MapComponent({
+  data,
+  userLocation,
+  permissionsStatus,
+  setPermissionsStatus,
+  handleMapPress,
+  handleMarkerPress,
+  focusedMarker,
+  focusedMarkerRef,
+  handleCalloutPress,
+  alertOnLocationError,
+}: Props) {
+  const {
+    theme: { colors, mode: themeMode },
+  } = useTheme()
+  const styles = useStyles()
+  const { LL } = useI18nContext()
+  const client = useApolloClient()
+  const text = LL.MapScreen.payBusiness()
+
+  const mapViewRef = useRef<MapView>(null)
+  const openSettingsModalRef = React.useRef<OpenSettingsElement>(null)
+
+  // toggle modal from inside modal component instead of here in the parent
+  const toggleModal = React.useCallback(
+    () => openSettingsModalRef.current?.toggleVisibility(),
+    [],
+  )
+
+  const respondToBlocked = (status: PermissionStatus) => {
+    // iOS will only ever ask once for permission, and initial checks can differentiate between BLOCKED vs DENIED
+    if (isIOS) {
+      if (permissionsStatus === RESULTS.BLOCKED && status === RESULTS.BLOCKED) {
+        toggleModal()
+      }
+      // Android can ask twice for permission, and initial checks cannot differentiate between BLOCKED vs DENIED
+    } else {
+      // TODO figure out how to not fire this after the 2nd time user blocks
+      toggleModal()
+    }
+  }
+
+  const centerOnUser = async () => {
+    getUserRegion(async (region) => {
+      if (region && mapViewRef.current) {
+        mapViewRef.current.animateToRegion(region)
+      } else {
+        alertOnLocationError()
+      }
+    })
+  }
+
+  const requestLocationPermission = async () => {
+    request(LOCATION_PERMISSION)
+      .then((status) => {
+        if (status === RESULTS.GRANTED) {
+          centerOnUser()
+        } else if (status === RESULTS.BLOCKED) {
+          respondToBlocked(status)
+        }
+        setPermissionsStatus(status)
+      })
+      .catch(() => alertOnLocationError())
```

Comment:
could we conver this to async/await syntax?
we don't really then().catch(), imo it's less readable than the new code syntax